### DISCUSSION TITLE
Missing classes when trying to compile USE with maven

### DISCUSSION THREAD

**Discussion by amielsaa** [useocl/use#63] [General]
Hello,
We are students from BGU, working with Mira and trying to start implementing the MLM (Multi Level Model).
We cloned USE repository and attempted to build it, but encountered some errors that said we were missing some classes.
I've attached screenshots of some of the errors that we are getting.

![image](https://user-images.githubusercontent.com/79751392/234918134-67546834-0c74-4704-9d46-6797176968d4.png)
![image](https://user-images.githubusercontent.com/79751392/234919208-6fa52f24-5631-4a55-acdc-c63e54c1ba9d.png)

What in your opinion may cause this error?
Thanks a lot!

**Comment by h-man2** [comment]
Hallo to BGU!
Did the maven build report any error? It is important that at least the compile lifecycle was used:
```
mvn compile
```
This generates the missing files above. 
A second try would be to resync the IDE after the maven build was executed.

Best regards

Lars

**Reply by Gil4390** [reply]
Thank you for your help!
running "mvn compile" before the build in the IDE indeed solved the issue

**Reply by ichxorya** [reply]
> Hallo to BGU! Did the maven build report any error? It is important that at least the compile lifecycle was used:
> 
> ```
> mvn compile
> ```
> 
> This generates the missing files above. A second try would be to resync the IDE after the maven build was executed.
> 
> Best regards
> 
> Lars

We should have a document on how to run and develop USE to avoid these issues.

**Reply by h-man2** [reply]
Like this? https://github.com/useocl/use/blob/master/INSTALL

**Reply by ichxorya** [reply]
> Like this? https://github.com/useocl/use/blob/master/INSTALL

Yes, but seems that the guide is insufficient, i.e. not mentioning how and why we need to do mvn compile (I figure it out myself to run mvn package, it's kinda the same - all is to get some required java files)
